### PR TITLE
Improve shell portability on NetBSD

### PR DESCRIPTION
**Pull request by krytarowski**
The "test" command, as well as the "[" command, are not required to know
the "==" operator.

Comparing with empty strings ("") might lead to unspecified results.

### CODE REVIEW

**Comment by akoeplinger** [comment]
build

File: mono/tests/verifier/make_tests.sh [hunk 1]
```diff
@@ -791,17 +791,17 @@ do
  	ZZ=`echo $TYPE2 | grep "*";`
  	T2_PTR=$?
  	
-    if (($T1_PTR == 0  ||  $T2_PTR == 0)); then
+    if (($T1_PTR = 0  ||  $T2_PTR = 0)); then
```
Comment (akoeplinger) [inline mono/tests/verifier/make_tests.sh]: should this be `-eq 0` too?

File: mono/tests/verifier/make_tests.sh [hunk 2]
```diff
@@ -3326,7 +3326,7 @@ done
 
 #branching to a protected block from the outside
 for I in {2..6}; do
-	if [ "$I" == "4" ]; then
+	if [ "$I" = "4" ]; then
```
Comment (akoeplinger) [inline mono/tests/verifier/make_tests.sh]: `-eq 4`?

File: mono/tests/verifier/make_tests.sh [hunk 3]
```diff
@@ -3362,7 +3362,7 @@ done
...
```
Comment (akoeplinger) [inline mono/tests/verifier/make_tests.sh]: `-eq 4`?

**Comment by krytarowski** [comment]
Well, for start I was focused on `==` with few other issues in this PR.

These script still aren't POSIX compliant. For example "$abc" can evaluate into an empty string and trigger undefined behavior with certain shells and needs proper handling ("x$abc").

There are more unportable constructs, like:

```
for I in {2..6}; do done
```

These loops should be replaced with:

```
i=2; while [ $i -le 6 ]; do ...; i=$((i + 1)); done
```

I was fixing the parts that break the mono build in pkgsrc, as there are scripts to detect part of portability problems and stops the build early in case of such problems. I will fix the parts that were mentioned by you and focus on making mono buildable and later runnable.

**Comment by krytarowski** [comment]
Done!

**Pull request merged**